atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractGetCandidateCellsForSweepingTest.java

```diff
@@ -98,19 +98,52 @@ public void reportLatestEmptyValue() {
                 .containsExactly(
                     ImmutableCandidateCellForSweeping.builder()
                         .cell(cell(1, 1))
-                        .sortedTimestamps(5L, 10L)
+                        .sortedTimestamps(ImmutableList.of(5L, 10L))
                         .isLatestValueEmpty(true)
-                        .numCellsTsPairsExamined(2)
                         .build(),
                     ImmutableCandidateCellForSweeping.builder()
                         .cell(cell(2, 2))
-                        .sortedTimestamps(4L, 9L)
+                        .sortedTimestamps(ImmutableList.of(4L, 9L))
                         .isLatestValueEmpty(false)
-                        .numCellsTsPairsExamined(4)
                         .build());
     }
 
     @Test
+    public void reportLatestEmptyValueLessThanSweepTs() {
+        new TestDataBuilder()
+                .put(1, 1, 5L)
+                .putEmpty(1, 1, 10L)
+                .putEmpty(2, 2, 5L)
+                .put(2, 2, 10L)
+                .store();
+        assertThat(getAllCandidates(thoroughRequest(PtBytes.EMPTY_BYTE_ARRAY, 8L, 100)))
+                .containsExactly(
+                        ImmutableCandidateCellForSweeping.builder()
+                                .cell(cell(1, 1))
+                                .sortedTimestamps(ImmutableList.of(5L))
+                                .isLatestValueEmpty(false)
+                                .build(),
+                        ImmutableCandidateCellForSweeping.builder()
+                                .cell(cell(2, 2))
+                                .sortedTimestamps(ImmutableList.of(5L))
+                                .isLatestValueEmpty(true)
+                                .build());
+    }
+
+    @Test
+    public void ignoresTimestampsToIgnore() {
+        new TestDataBuilder()
+                .put(1, 1, -1L)
+                .put(1, 1, 5L)
+                .putEmpty(2, 2, -1L)
+                .put(2, 2, 10L)
+                .store();
+        boolean containsIgnoredTimestamps = getAllCandidates(conservativeRequest(PtBytes.EMPTY_BYTE_ARRAY, 100L, 100))
+                .stream().anyMatch(candidate -> candidate.sortedTimestamps().contains(-1L));
```

Comment:
completely inconsequential nit: You can also use `.noneMatch`, if you want to assert that it's true.